### PR TITLE
Fix issue with settle() asyncFee is left in the KeepFactory and is not transfer to the keeper

### PR DESCRIPTION
**Pull request by prateekdefi**
Issue: https://github.com/sherlock-audit/2024-08-perennial-v2-update-3-judging/issues/34

### CODE REVIEW

**Comment by prateekdefi** [comment]
@EdNoepel  @kbrizzle  @arjun-io. We can also do this by passing the keeper address from keeper factory to keeper oracle. That would be more gas efficient(only a single push) but it would change the logic for both the factory and the oracle. What do u guys think is a better option.

**Review by EdNoepel — COMMENTED** [review]
Perhaps we should add a test which settles and checks keeper balance to ensure they got the fee?  Such a test should have failed with the old implementation.

File: packages/perennial-oracle/contracts/keeper/KeeperFactory.sol [hunk 1]
```diff
@@ -164,8 +164,10 @@ abstract contract KeeperFactory is IKeeperFactory, Factory {
         if (oracleIds.length == 0 || oracleIds.length != versions.length || oracleIds.length != maxCounts.length)
             revert KeeperFactoryInvalidSettleError();
 
-        for (uint256 i; i < oracleIds.length; i++)
-            IKeeperOracle(address(oracles[oracleIds[i]])).settle(versions[i], maxCounts[i]);
+        for (uint256 i; i < oracleIds.length; i++) {
+            Token18 rewardToken = IKeeperOracle(address(oracles[oracleIds[i]])).oracle().market().token();
```
Comment (kbrizzle) [inline packages/perennial-oracle/contracts/keeper/KeeperFactory.sol:168]: i think this line is unused, we just need to pass the new receiver in

**Comment by kbrizzle** [comment]
As @EdNoepel mentioned, would be good to have an additional test that fails on the old implementation.

File: packages/perennial-oracle/contracts/keeper/KeeperFactory.sol (hunk 1 above)
Comment (EdNoepel) [reply packages/perennial-oracle/contracts/keeper/KeeperFactory.sol]: removed in f55c199ad350bf0f8f08e316444c298a68aff19a

**Comment by EdNoepel** [comment]
> As @EdNoepel mentioned, would be good to have an additional test that fails on the old implementation.

Upon review, this is exactly what `settles successfully and incentivizes the keeper` was testing.  Just the amount was wrong.  Attempted to calculate it, but the fee depends on gas consumption and such.

**Review by arjun-io — APPROVED** [review]

**Pull request merged**